cdap-credential-ext-gcp-wi/src/main/java/io/cdap/cdap/security/spi/credential/GcpWorkloadIdentityCredentialProvider.java

```diff
@@ -185,8 +206,17 @@ public ProvisionedCredential provision(NamespaceMeta namespaceMeta,
     try {
       while (stopWatch.elapsed(TimeUnit.SECONDS) < timeout) {
         try {
-          return getCredentialLoadingCache().get(new ProvisionedCredentialCacheKey(namespaceMeta,
-              identity, scopes));
+          // Get k8s namespace from namespace metadata if using a non-default namespace and
+          // namespace creation hook is enabled.
+          String k8sNamespace = NamespaceId.DEFAULT.getNamespace();
+          String cdapWrappedNamespace = context.getProperties().get(
+              GCP_WRAPPED_CDAP_NAMESPACE_PROPERTY);
+          if (!cdapWrappedNamespace.equals(NamespaceId.DEFAULT.getNamespace())
+              && credentialProviderContext.isNamespaceCreationHookEnabled()) {
+            k8sNamespace = context.getProperties().get(K8S_NAMESPACE_PROPERTY);
+          }
+          return getCredentialLoadingCache().get(new ProvisionedCredentialCacheKey(k8sNamespace,
```

Comment:
`cdapWrappedNamespace` and not `k8sNamespace` which will be default always for non-hybrid mode.

Reply:
Discussed offline to include identity in cache key.